### PR TITLE
dev mode components / routes with castle risk fake page

### PR DESCRIPTION
**Pull request by tommysullivan**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -59,12 +71,18 @@ const facebookUserSessionCookieKey = 'facebookUserSession';
 const deleteSessionCookie = () => Cookies.remove(facebookUserSessionCookieKey);
 const facebookUserSessionString = Cookies.get(facebookUserSessionCookieKey);
 const facebookUserSession = facebookUserSessionString && JSON.parse(facebookUserSessionString);
+const isLocalhost = () => window.location.hostname == "localhost";
+const developers = ['tom@tommysullivan.me','mrcorn123@yahoo.com','tastulae@mail.usf.edu',"aashreya.isforever@gmail.com"]
```
Comment (bob-bins) [inline src/index.tsx]: add peepcake

File: tsconfig.json [hunk 2]
```diff
@@ -37,7 +37,12 @@
 
     /* Module Resolution Options */
     // "moduleResolution": "node",            /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
-    // "baseUrl": "./",                       /* Base directory to resolve non-absolute module names. */
+    
+    
+    //TODO: Make this work (it works for VS Code, but webpacked runtime in browser yields errors, haven't checked server code)
```
Comment (bob-bins) [inline tsconfig.json]: 😮

File: src/utils/Switch.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+export const Switch = <T, S>(expression:T, ...cases:Array<CaseLike<T, S>>):S => {
+    const firstMatchingCase = cases.find(c => c.matches(expression));
+    if(!firstMatchingCase) throw new Error(`No cases matched expression with value ${expression}`);
```
Comment (aashreya) [inline src/utils/Switch.ts]: Just curious if we have this condition we don't get to the default case right ? Maybe I am not reading this correctly.

Comment (tommysullivan) [reply src/utils/Switch.ts]: the default case is one of the cases, it is the case that, if it exists, it always matches. so the only time we would hit this is if we had a switch where none of the cases matched and there was no default case.

File: src/index.tsx (hunk 1 above)
Comment (aashreya) [reply src/index.tsx]: Yup that is my email.

**Pull request merged**